### PR TITLE
SDK release for 2017-06-01.5.1.

### PR DESCRIPTION
**Pull request by xingwu1**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ var fs = require('fs');
 
 var mappings = {
     'batchService': {
-        'source': 'batch/2017-05-01.5.0/swagger/BatchService.json',
+        'source': 'specification/batch/data-plane/Microsoft.Batch/2017-06-01.5.1/BatchService.json',
```
Comment (matthchr) [inline gulpfile.js]: Does it make sense to reference an individual commit here?

File: src/test/java/com/microsoft/azure/batch/PoolTests.java [hunk 2]
```diff
@@ -67,8 +67,30 @@ public void canCRUDLowPriIaaSPool() throws Exception {
...
+            netConfig.withEndpointConfiguration(endpointConfig);
+
+            PoolAddParameter addParameter = new PoolAddParameter()
+                    .withId(poolId)
+                    .withTargetDedicatedNodes(POOL_VM_COUNT)
+                    .withTargetLowPriorityNodes(POOL_LOW_PRI_VM_COUNT)
+                    .withVmSize(POOL_VM_SIZE)
+                    .withVirtualMachineConfiguration(configuration)
+                    .withNetworkConfiguration(netConfig);
+                    //.withApplicationLicenses(licenses);
```
Comment (matthchr) [inline src/test/java/com/microsoft/azure/batch/PoolTests.java]: Remove this?

File: src/test/java/com/microsoft/azure/batch/TaskTests.java [hunk 3]
```diff
@@ -282,7 +282,12 @@ public void testAddMultiTasks() throws Exception {
             // LIST
             List<CloudTask> tasks = batchClient.taskOperations().listTasks(jobId);
             Assert.assertNotNull(tasks);
-            Assert.assertTrue(tasks.size() == 1000);
+            Assert.assertTrue(tasks.size() == TASK_COUNT);
+
+            // Test Job count
+            TaskCounts counts = batchClient.jobOperations().getTaskCounts(jobId);
+            int all = counts.active() + counts.completed() + counts.running();
```
Comment (matthchr) [inline src/test/java/com/microsoft/azure/batch/TaskTests.java]: Assert on the `validationState` too?

Also might be helpful to have this in a different test than the `AddMultiTasks` one just for clarity/separation of concerns?

File: README.md [hunk 4]
```diff
@@ -51,25 +60,25 @@ You can find sample code that illustrates Batch usage scenarios in https://githu
...
     <groupId>com.microsoft.azure</groupId>
     <artifactId>azure-batch</artifactId>
-    <version>1.0.0</version>
+    <version>2.1.0</version>
 </dependency>
 <dependency>
     <groupId>com.microsoft.rest</groupId>
     <artifactId>client-runtime</artifactId>
-    <version>1.0.0</version>
+    <version>1.0.1</version>
```
Comment (jianghaolu) [inline README.md]: Recommend using 1.0.3+ since there were a couple critical bug fixes.

File: README.md [hunk 5]
```diff
@@ -51,25 +60,25 @@ You can find sample code that illustrates Batch usage scenarios in https://githu
...
 </dependency>
 <dependency>
     <groupId>com.microsoft.azure</groupId>
     <artifactId>azure-client-runtime</artifactId>
-    <version>1.0.0</version>
+    <version>1.0.1</version>
```
Comment (jianghaolu) [inline README.md]: Same here.

Comment (jianghaolu) [inline README.md]: Please reflect these versions in your pom as well.

File: gulpfile.js (hunk 1 above)
Comment (xingwu1) [inline gulpfile.js]: This repo is only for azure batch sdk. So I think this is fine to just reference to batch swagger spec.

File: README.md (hunk 5 above)
Comment (xingwu1) [inline README.md]: fixed.

File: README.md (hunk 4 above)
Comment (xingwu1) [inline README.md]: fixed.

File: src/test/java/com/microsoft/azure/batch/PoolTests.java (hunk 2 above)
Comment (xingwu1) [inline src/test/java/com/microsoft/azure/batch/PoolTests.java]: Done.

File: src/test/java/com/microsoft/azure/batch/TaskTests.java (hunk 3 above)
Comment (xingwu1) [inline src/test/java/com/microsoft/azure/batch/TaskTests.java]: Add validationState validation.

Changed to 2 tests.

If separate the two tests, then the code have to du

**Pull request merged**